### ISSUE TITLE
String as input in petastorm dataloaders

### ISSUE THREAD

**Issue by freud14-tm** [uber/petastorm#795]
In the `_sanitize_pytorch_types` function used in petastorm dataloaders, an exception is raised when an input is a string:

https://github.com/uber/petastorm/blob/0b0775af42539189d913702d9695566431dabd8a/petastorm/pytorch.py#L64

Is this really necessary? Maybe the [`default_collate`](https://pytorch.org/docs/stable/data.html#torch.utils.data.default_collate) function from PyTorch might not always like, although it seems to support it according to the documentation, but this prevents making other types of collate functions.

Would you consider removing the exception? Thank you.

**Comment by selitvin** [comment]
Sure! Would you like to propose a PR for that? Ideally, with a small unit test showing that it works.

**Comment by freud14-tm** [comment]
It seems more complicated than I thought with the current code base of the library and I am not using the library right now. Still think the idea is worthwhile but feel free to close the issue if you wish. Otherwise, let it open in order to keep the idea close.

**Comment by pratikgujjar** [comment]
Wondering if there is a reason for the string type to not be supported? A column in a parquet file could be all string labels. It is currently not possible to read this parquet file as a PyTorch dataloader becuase this line in the code.
